lib/resilient/circuit_breaker.rb

```diff
@@ -1,9 +1,33 @@
 require "resilient/key"
 require "resilient/circuit_breaker/metrics"
 require "resilient/circuit_breaker/properties"
+require "resilient/circuit_breaker/registry"
 
 module Resilient
   class CircuitBreaker
+    # Public: Resets all registered circuit breakers (and thus there state/metrics).
+    # Useful for ensuring a clean environment between tests.
+    def self.reset
+      registry.reset
+    end
+
+    # Private: Registry that stores instances of circuit breakers to ensure that
+    # attempts to create new instances with the same key retain existing state/metrics.
+    def self.registry
+      @registry ||= Registry.new
+    end
+    class << self; private :registry; end
+
+    # Public: Overriden method to return instance of circuit breaker based on
+    # key. If key does not exist, it is registered. If key does exist, it
+    # returns registered instance instead of allocating a new instance in order
+    # to ensure that state/metrics are the same per key.
+    def self.new(key: nil, open: false, properties: nil, metrics: nil)
```

Comment:
I went back and forth on this. It felt better to override new than to force people to use a new method (CircuitBreaker.instance or something). Forcing a different method (and not privatizing new) could have led to bugs where people thought they were using the same circuit based on key, but weren't (if they forgot to use the different method and used new instead).